tests/http_server.py

```diff
@@ -22,5 +22,10 @@ async def get():
     return "hi"
 
 
+@app.get("/robots.txt")
```

Comment:
Won’t this change break tests for x-robots-header parsing by causing the downloader to fail early?  Maybe make a different 3rd mount point that’s disallowed by robots.txt than the one that’s disallowed by http headers?

Reply:
it would've, but that test doesn't actually test whether the disallowed paths/files are not downloaded or what point in the process they were disallowed. The FastAPI server was returning a json response for robots.txt causing it to be ignored (content type was meant to be text/plain).

The test case still doesn't validate that the disallow method was correct but will at least fail if one isn't working now.